state/prices/selectors.ts

```diff
@@ -4,22 +4,47 @@ import { wei } from '@synthetixio/wei';
 import { Prices } from 'sdk/types/prices';
 import { RootState } from 'state/store';
 import { getPricesForCurrencies } from 'utils/currencies';
+import { priceChangeToColor } from 'utils/prices';
+
+import { PriceColors } from './types';
 
 export const selectPrices = createSelector(
 	(state: RootState) => state.prices,
 	({ onChainPrices, offChainPrices }) => {
 		const merged: Prices = {};
-		Object.entries(onChainPrices).forEach(([key, value]) => {
+		Object.entries(onChainPrices).forEach(([key, { price }]) => {
+			merged[key] = {
+				onChain: wei(price),
+			};
+		});
+		Object.entries(offChainPrices).forEach(([key, { price }]) => {
+			if (merged[key]) {
+				merged[key].offChain = wei(price);
+			} else {
+				merged[key] = {
+					offChain: wei(price),
+				};
+			}
+		});
+		return merged;
+	}
+);
+
+export const selectPriceColors = createSelector(
```

Comment:
I'm not sure we need this extra selector, we could just have the `ColoredPrice` label accept a `PriceInfo` object and it decide the color based on change?

Reply:
This works, but I had to make a function to export for `MarketDetails` to convert `PricesInfo` into a color string since we don't use that component.